pkg/v1/mutate/mutate.go

```diff
@@ -101,6 +101,23 @@ func RemoveManifests(base v1.ImageIndex, matcher match.Matcher) v1.ImageIndex {
 	}
 }
 
+// RemoveAnnotations removes annotations the provided v1.Image
+func RemoveAnnotations(base v1.Image, annotations []string) (v1.Image, error) {
```

Comment:
Don't return an error here, we can do this lazily. We'll need to store this list of annotations to delete in image{} (maybe call it `unannotations` and call `delete()` around here: https://github.com/google/go-containerregistry/blob/eca1cd86affe459c22a2c7a818769ea70d13496e/pkg/v1/mutate/image.go#L157

I also wonder if we want to give this the same treatment as https://github.com/google/go-containerregistry/pull/1082

Reply:
That makes sense! Now, it's much cleaner. 👍